commitments/mode.go

```diff
@@ -31,19 +31,23 @@ func StringToCommitmentMode(s string) (CommitmentMode, error) {
 	}
 }
 
-func EncodeCommitment(b []byte, c CommitmentMode) ([]byte, error) {
-	switch c {
+func EncodeCommitment(
+	bytes []byte,
+	commitmentMode CommitmentMode,
+	commitmentType EigenDACommitmentType,
+) ([]byte, error) {
+	switch commitmentMode {
 	case OptimismKeccak:
-		return Keccak256Commitment(b).Encode(), nil
+		return Keccak256Commitment(bytes).Encode(), nil
 
 	case OptimismGeneric:
-		certCommit := NewV0CertCommitment(b).Encode()
+		certCommit := NewEigenDACommitment(bytes, commitmentType).Encode()
 		svcCommit := EigenDASvcCommitment(certCommit).Encode()
 		altDACommit := NewGenericCommitment(svcCommit).Encode()
 		return altDACommit, nil
 
-	case Standard:
-		return NewV0CertCommitment(b).Encode(), nil
+	case Standard: // (i.e, Arbitrum)
```

Comment:
Think we'll need to stop being arbitrum specific for the standard commitment mode, especially now that AVSs want to also start using that route. I'm guessing other rollup stacks would as well. So Standard is basically everyone but OP.

Reply:
Removed this comment [0f7551f](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/0f7551f013acee9065018405807b502d4cd07f2b)